src/bpmncwpverify/bpmn.py

```diff
@@ -36,18 +55,35 @@ class StartEvent(Event):
     def __init__(self, element: Element):
         super().__init__(element)
 
+    def accept(self, visitor: "BpmnVisitor") -> None:
+        result = visitor.visit_start_event(self)
+        self.visit_out_flows(visitor, result)
+        visitor.end_visit_start_event(self)
+
 
 class EndEvent(Event):
     def __init__(self, element: Element):
         super().__init__(element)
 
+    def accept(self, visitor: "BpmnVisitor") -> None:
+        result = visitor.visit_end_event(self)
+        self.visit_out_flows(visitor, result)
+        visitor.end_visit_end_event(self)
+
 
 class IntermediateEvent(Event):
     def __init__(self, element: Element):
         super().__init__(element)
 
+    def accept(self, visitor: "BpmnVisitor") -> None:
+        result = visitor.visit_intermediate_event(self)
+        self.visit_out_flows(visitor, result)
```

Comment:
Let's rename `visit_out_flows`. It's too confusing to mix the meaning of `visit` with the meaning of `accept`. Yes. Technically, call the `accept` on each outflow is a `visit`, but the `visit` term belongs to the `Visitor` while the `accept` term belongs to traversal. Maybe `travers_out_flows`? The other confusing thing is that it is conditional. So maybe `traverse_outflows_if_result`?

Reply:
100%, that is really confusing.